equinox/nn/_embedding.py

```diff
@@ -194,12 +201,14 @@ def precompute_freqs_cis(
     def __call__(
         self,
         x: Float[Array, "seq_length embedding_size"],
+        offset: Int[Array, ""] = jnp.array(0),
```

Comment:
Ah, I didn't know. Is this just Equinox specific or should I in general avoid JAX arrays for defaults? 

In any case, I made this mandatory then and added some more examples. Unfortunately, something like:

```python
offset: Optional[Int[Array, ""]] = None

offset = offset if offset is not None else jnp.array(0)
...

doesn't work under JIT and `jax.lax.select` doesn't work either because it evaluates both branches. 

You don't happen to have a solution for these kind of issues somewhere in your JAX wizard hat? Understanding JAX's behaviour under JIT is definitely something I need to catch up on!